modules/user-access/pages/access-control-model.adoc

```diff
@@ -20,25 +21,127 @@ This allows the user to run commands such as `ls` and `SHOW VERTEX` on the graph
 
-Some privileges can only be global by nature. For example, since users are global objects, any user-related privileges are global only. To see which privileges are global-only, see xref:reference:list-of-privileges.adoc[List of Privileges].
+The privileges govern specific ways in which a user can access and modify data.
+Even regarding a single REST endpoint, whether a request is authorized depends on the information that the request is accessing or modifying.
+
+|===
+|Privilege |Type level |Attribute level
+
+|`CREATE_DATA`
+|Permission to create vertices and edges, and specify values for all attributes for the type where the privilege is granted.
+a|* If granted on a vertex type attribute, it gives permission to create vertices, but only specify values for attributes where the user has `CREATE_DATA` privilege if the user also has `UPDATE_DATA` privilege on all attributes of that type.
+** The user must have `CREATE_DATA` privilege on the primary ID of the vertex type to be able to create vertices.
+* If granted on an edge type attribute, it gives permission to create edges, but only specify values for attributes where the user has `CREATE_DATA` privilege if the user also has `UPDATE_DATA` privilege on all attributes of that type.
+* For attributes where the user doesn't have privilege, they must use wildcards(`_`) to use the default value for vertices/edges created by the xref:gsql-ref:querying:data-modification-statements.adoc#_insert_into_statement[`INSERT INTO` statements].
+
+|`READ_DATA`
+|Permission to access all data of the type where the privilege is granted.
+|Permission to access the attribute value where the privilege is granted.
+
+To grant  `READ_DATA` to a specific attribute of a type, you must grant `READ_DATA` to the primary key of the type first or in the same command.
+For edges, you must grant `READ_DATA` to the primary key of the `FROM` and `TO`
+vertex types before granting `READ_DATA` to other attributes of the edge type.
+|`UPDATE_DATA`
+|Permission to update all data of the type where the privilege is granted.
+|Permission to update the attribute value where the privilege is granted.
+`UPDATE_DATA` on all attributes is also required for creating new vertices and edges.
+
+|`DELETE_DATA`
+|Permission to delete data of the type where the privilege is granted.
+|N/A.
+This privilege is not applicable on the attribute level.
+|===
+
+==== Examples
+
+Suppose we have a graph with schema as below:
+
+[.wrap,gsql]
+----
+CREATE VERTEX Person(id UINT PRIMARY KEY, name STRING, INT age)
+CREATE VERTEX City(id UINT PRIMARY KEY, name STRING)
+CREATE GRAPH Example_Graph(Person, City)
+----
+
+If a user were to run the following query:
+
+[.wrap,gsql]
+----
+CREATE QUERY example_query() {
+  Seed = {City.*}; <1>
+  vSet = SELECT s FROM Seed:s
+         POST-ACCUM
+            s.name = s.name + ".post"; <2>
+}
```

Comment:
good example. Can we also add statement for `insert` and `delete`
Also, it'll be better if we can show the example of error message, from both GSQL and rest server if the user doesn't have enough privilege to run the query.

Reply:
Also highlight that for request to rest server must have rest auth enabled. Otherwise the rest server won't be able to check permission.

Reply:
Error messages are good to have in the "Error code to description format" for reference.  
usually, it's not a good idea to display error messages in the documentation about concepts and procedures. it's information that users will see when they interact with the product - so there is no need to duplicate the information. Also, if we ever update the error message, we'll almost certainly forget to update it in the docs, and the user will start to doubt if they are hitting a different error instea.d